src/config.h

```diff
@@ -40,8 +40,12 @@
 #include <fcntl.h>
 #endif
 
+#if defined(__APPLE__) && defined(__MAC_OS_X_VERSION_MAX_ALLOWED) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 1060
```

Comment:
Please add some comments to this to explain why we are doing this, maybe refer to the official documentation.

Reply:
please also apply [sundb/redis@5a3b7aa/.github/workflows/daily.yml#L33-L60](https://github.com/sundb/redis/blob/5a3b7aaaf9b4f04705c26c1affc545fe768d5ee6/.github/workflows/daily.yml#L33-L60) to this PR, and put it after `test-macos-latest-cluster` block.

Reply:
@ygcaicn Looks good, please finish the rest before merging.

Reply:
> Please add some comments to this to explain why we are doing this, maybe refer to the official documentation.

Please add a comment mentioning the detailed reasoning behind this.

Reply:
??